python/src/trezorlib/cli/firmware.py

```diff
@@ -102,20 +176,53 @@ def find_best_firmware_version(
     else:
         releases = releases_stable
 
-    releases.sort(key=lambda r: r["version"], reverse=True)
+    return releases
+
+
+def find_best_firmware_version(
+    client: TrezorClient,
+    version: str = None,
+    beta: bool = False,
+    bitcoin_only: bool = False,
+) -> tuple:
+    """Get the url from which to download the firmware and its expected fingerprint.
+
+    When the version (X.Y.Z) is specified, checks for that specific release.
+    Otherwise takes the latest one.
+
+    If the specified version is not found, prints the closest available version
+    (higher than the specified one, if existing).
+    """
 
     def version_str(version):
         return ".".join(map(str, version))
 
-    want_version = requested_version
+    f = client.features
+
+    releases = get_all_firmware_releases(bitcoin_only, beta, f.major_version)
+    releases.sort(key=lambda r: r["version"], reverse=True)
```

Comment:
please return the releases already sorted

Reply:
At first I had it like that, but then it seemed to me it could be confusing to implicitly sort, unless the function name was explicitly named "get_all_firmware_releases_sorted" or something like that (or the variable name "releases_sorted")

Reply:
Changed in 2ae4